state-chain/primitives/src/lib.rs

```diff
@@ -95,6 +95,20 @@ pub type BasisPoints = u16;
 
 pub type BroadcastId = u32;
 
+// TODO: Consider alternative representation for Price:
+//
+// increasing Price to U512 or switch to a f64 (f64 would only be for the external
+// price representation), as at low ticks the precision in the price is VERY LOW, but this does not
+// cause any problems for the AMM code in terms of correctness
+
+/// This is the ratio of equivalently valued amounts of asset One and asset Zero.
+///
+/// The price is always measured in amount of asset One per unit of asset Zero. Therefore as asset
+/// zero becomes more valuable relative to asset one the price's literal value goes up, and vice
+/// versa. This ratio is represented as a fixed point number with `PRICE_FRACTIONAL_BITS` fractional
+/// bits.
+pub type Price = U256;
```

Comment:
I've mentioned this many times - i'm not a fan of exposing a type alias like this as a 'primitive' when it's loaded with so much meaning...

I guess this is better than duplicating the definition though, so 🤷 

Created PRO-2044 to address this once and for all.